### PR TITLE
NEW - SMBClientAccessToServer cmdlet

### PR DESCRIPTION
**Pull request by Xelu86**
# PR Summary

This is a new SMB cmdlet, SMBClientAccessToServer, for WS25.

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: docset/winserver2025-ps/smbshare/Block-SmbClientAccessToServer.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+# Block-SmbClientAccessToServer
+
+## SYNOPSIS
+Blocks SMB client access to a specified server.
+
+## SYNTAX
+
+```
+Block-SmbClientAccessToServer [-Name] <String[]> -IdentifierType <IdentifierType>
+-Identifier <String> [-Description <String>] [-Force] [-CimSession <CimSession[]>]
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Block-SmbClientAccessToServer.md:20]: A space is required after the first line.
```suggestion
 -Identifier <String> [-Description <String>] [-Force] [-CimSession <CimSession[]>]
```

File: docset/winserver2025-ps/smbshare/Block-SmbClientAccessToServer.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -CimSession
+
+Runs the cmdlet in a remote session or on a remote computer. Enter a computer name or a session
+object, such as the output of a [New-CimSession](/powershell/module/cimcmdlets/new-cimsession) or
+[Get-CimSession](https://go.microsoft.com/fwlink/p/?LinkId=227966) cmdlet. The default is the
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Block-SmbClientAccessToServer.md:78]:
```suggestion
[Get-CimSession](/powershell/module/cimcmdlets/get-cimsession) cmdlet. The default is the
```

File: docset/winserver2025-ps/smbshare/Get-SmbClientAccessToServer.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -CimSession
+
+Runs the cmdlet in a remote session or on a remote computer. Enter a computer name or a session
+object, such as the output of a [New-CimSession](/powershell/module/cimcmdlets/new-cimsession) or
+[Get-CimSession](https://go.microsoft.com/fwlink/p/?LinkId=227966) cmdlet. The default is the
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Get-SmbClientAccessToServer.md:67]:
```suggestion
[Get-CimSession](/powershell/module/cimcmdlets/get-cimsession) cmdlet. The default is the
```

File: docset/winserver2025-ps/smbshare/Grant-SmbClientAccessToServer.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+    Name = "Server01"
+    IdentifierType = "ISSUER"
+    Identifier = "CN=MyCertificateIssuer"
+    Description = "Allow access from client for LabSvr"
+    Force = $true
+}
+Grant-SmbClientAccessToServer @params
+```
+
+This example grants SMB client access to a server named "Server01" based on the client's
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Grant-SmbClientAccessToServer.md:52]: Put parameter values in backticks.
```suggestion
This example grants SMB client access to a server named `Server01` based on the client's
```

File: docset/winserver2025-ps/smbshare/Grant-SmbClientAccessToServer.md [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -CimSession
+
+Runs the cmdlet in a remote session or on a remote computer. Enter a computer name or a session
+object, such as the output of a [New-CimSession](/powershell/module/cimcmdlets/new-cimsession) or
+[Get-CimSession](https://go.microsoft.com/fwlink/p/?LinkId=227966) cmdlet. The default is the
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Grant-SmbClientAccessToServer.md:79]:
```suggestion
[Get-CimSession](/powershell/module/cimcmdlets/get-cimsession) cmdlet. The default is the
```

File: docset/winserver2025-ps/smbshare/Grant-SmbClientAccessToServer.md [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Identifier
+
+Specifies the identity of the client that is being granted access to the SMB server. This parameter
+takes a string value that represents the identity of the client. The format of the string value
+will depend on the `-IdentifierType` parameter that you're using.
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Grant-SmbClientAccessToServer.md:149]: In paragraphs, use backticks for syntax examples; bold for parameter names.
```suggestion
will depend on the **IdentifierType** parameter that you're using.
```

File: docset/winserver2025-ps/smbshare/Grant-SmbClientAccessToServer.md [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -IdentifierType
+
+Specifies the type of identifier that you're using to identify a client. The valid values for the
+`-IdentifierType` parameter are:
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Grant-SmbClientAccessToServer.md:166]:
```suggestion
**IdentifierType** parameter are:
```

File: docset/winserver2025-ps/smbshare/Revoke-SmbClientAccessToServer.md [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -CimSession
+
+Runs the cmdlet in a remote session or on a remote computer. Enter a computer name or a session
+object, such as the output of a [New-CimSession](/powershell/module/cimcmdlets/new-cimsession) or
+[Get-CimSession](https://go.microsoft.com/fwlink/p/?LinkId=227966) cmdlet. The default is the
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Revoke-SmbClientAccessToServer.md:71]:
```suggestion
[Get-CimSession](/powershell/module/cimcmdlets/get-cimsession) cmdlet. The default is the
```

File: docset/winserver2025-ps/smbshare/Revoke-SmbClientAccessToServer.md [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Identifier
+
+Specifies the identity of the client that is being granted access to the SMB server. This parameter
+takes a string value that represents the identity of the client. The format of the string value
+will depend on the `-IdentifierType` parameter that you're using.
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Revoke-SmbClientAccessToServer.md:122]:
```suggestion
will depend on the **IdentifierType** parameter that you're using.
```

File: docset/winserver2025-ps/smbshare/Revoke-SmbClientAccessToServer.md [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -IdentifierType
+
+Specifies the type of identifier that you're using to identify a client. The valid values for the
+`-IdentifierType` parameter are:
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Revoke-SmbClientAccessToServer.md:139]:
```suggestion
**IdentifierType** parameter are:
```

File: docset/winserver2025-ps/smbshare/Unblock-SmbClientAccessToServer.md [hunk 11]
```diff
@@ -0,0 +1,265 @@
...
+    Name = "Server01"
+    IdentifierType = "ISSUER"
+    Identifier = "CN=MyCertificateIssuer"
+    Force = $true
+}
+Unblock-SmbClientAccessToServer @params
+```
+
+This command unblocks SMB client access to a server named "Server01" for the based on the client's
+certificate. This command runs without prompting for confirmation as the `-Force` parameter is set
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Unblock-SmbClientAccessToServer.md:52]:
```suggestion
certificate. This command runs without prompting for confirmation as the **Force** parameter is set
```

File: docset/winserver2025-ps/smbshare/Unblock-SmbClientAccessToServer.md [hunk 12]
```diff
@@ -0,0 +1,265 @@
...
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -CimSession
+
+Runs the cmdlet in a remote session or on a remote computer. Enter a computer name or a session
+object, such as the output of a [New-CimSession](/powershell/module/cimcmdlets/new-cimsession) or
+[Get-CimSession](https://go.microsoft.com/fwlink/p/?LinkId=227966) cmdlet. The default is the
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Unblock-SmbClientAccessToServer.md:78]:
```suggestion
[Get-CimSession](/powershell/module/cimcmdlets/get-cimsession) cmdlet. The default is the
```

**Review by sdwheeler — CHANGES_REQUESTED** [review]
See my suggested changes.

File: docset/winserver2025-ps/smbshare/Block-SmbClientAccessToServer.md [hunk 13]
```diff
@@ -0,0 +1,284 @@
...
+
+# Block-SmbClientAccessToServer
+
+## SYNOPSIS
+Blocks SMB client access to a specified server.
+
+## SYNTAX
+
+```
+Block-SmbClientAccessToServer [-Name] <String[]> -IdentifierType <IdentifierType>
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Block-SmbClientAccessToServer.md:19]: You are missing the H3 parameter set names for both parameter blocks.

File: docset/winserver2025-ps/smbshare/Get-SmbClientAccessToServer.md [hunk 14]
```diff
@@ -0,0 +1,195 @@
...
+
+```
+Get-SmbClientAccessToServer [-Name] <String[]> [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>]
+ [-AsJob] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+```
+Get-SmbClientAccessToServer -InputObject <CimInstance[]> [-CimSession <CimSession[]>]
+ [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Get-SmbClientAccessToServer.md:26]: Missing parameter set names.

File: docset/winserver2025-ps/smbshare/Get-SmbClientAccessToServer.md [hunk 15]
```diff
@@ -0,0 +1,195 @@
...
+
+### -Name
+
+Specifies a fully-qualified DNS name or NetBIOS name that must match the certificate's subject name
+or an entry in the certificate's subject alternative names.
+
+```yaml
+Type: String[]
+Parameter Sets: (All)
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Get-SmbClientAccessToServer.md:121]: Incorrect parameter set

File: docset/winserver2025-ps/smbshare/Get-SmbClientAccessToServer.md [hunk 16]
```diff
@@ -0,0 +1,195 @@
...
+Type: SwitchParameter
+Parameter Sets: (All)
+Aliases: cf
+
+Required: False
+Position: Named
+Default value: False
+Accept pipeline input: False
+Accept wildcard characters: False
+```
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Get-SmbClientAccessToServer.md:96]: **Confirm** and **WhatIf** parameters go at the end of the list before CommonParameters.

File: docset/winserver2025-ps/smbshare/Get-SmbClientAccessToServer.md [hunk 17]
```diff
@@ -0,0 +1,195 @@
...
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Confirm
+
+Prompts you for confirmation before running the cmdlet.
+
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Get-SmbClientAccessToServer.md:85]: **Confirm** and **WhatIf** parameters go at the end of the list before CommonParameters.

File: docset/winserver2025-ps/smbshare/Grant-SmbClientAccessToServer.md [hunk 18]
```diff
@@ -0,0 +1,285 @@
...
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Confirm
+
+Prompts you for confirmation before running the cmdlet.
+
+```yaml
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Grant-SmbClientAccessToServer.md:98]: **Confirm** and **WhatIf** parameters go at the end of the list before CommonParameters.

File: docset/winserver2025-ps/smbshare/Revoke-SmbClientAccessToServer.md [hunk 19]
```diff
@@ -0,0 +1,258 @@
...
+```
+
+### -Confirm
+
+Prompts you for confirmation before running the cmdlet.
+
+```yaml
+Type: SwitchParameter
+Parameter Sets: (All)
+Aliases: cf
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Revoke-SmbClientAccessToServer.md:93]: **Confirm** and **WhatIf** parameters go at the end of the list before CommonParameters.

File: docset/winserver2025-ps/smbshare/Unblock-SmbClientAccessToServer.md [hunk 20]
```diff
@@ -0,0 +1,265 @@
...
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Confirm
+
+Prompts you for confirmation before running the cmdlet.
+
+```yaml
```
Comment (sdwheeler) [inline docset/winserver2025-ps/smbshare/Unblock-SmbClientAccessToServer.md:97]: **Confirm** and **WhatIf** parameters go at the end of the list before CommonParameters.

**Review by sdwheeler — COMMENTED** [review]
Fix the order of **Confirm** and **WhatIf** parameters. Add Parameter Set names. Update the parameters with the proper parameter set names.

File: docset/winserver2025-ps/smbshare/Block-SmbClientAccessToServer.md [hunk 21]
```diff
@@ -0,0 +1,288 @@
...
+
+```
+Block-SmbClientAccessToServer -InputObject <CimInstance[]> -IdentifierType <IdentifierType>
+ -Identifier <String> [-Description <String>] [-Force] [-CimSession <CimSession[]>]
+ [-ThrottleLimit <Int32>]  [-AsJob] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+## DESCRIPTION
+
+The `Block-SmbClientAccessToServer` cmdlet is used to prevent SMB clients from accessing a
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/Block-SmbClientAccessToServer.md:36]: When using SMB over QUIC client access control

File: docset/winserver2025-ps/smbshare/Get-SmbClientAccessToServer.md [hunk 22]
```diff
@@ -0,0 +1,199 @@
...
+```
+Get-SmbClientAccessToServer -InputObject <CimInstance[]> [-CimSession <CimSession[]>]
+ [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+## DESCRIPTION
+
+The `Get-SmbClientAccessToServer` cmdlet is used to retrieve information about SMB client access to
+a specified server. When you run this cmdlet, it'll return a list of all SMB client connections to
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/Get-SmbClientAccessToServer.md:35]: not a list of connections, but a list of allowed and blocked connections for SMB over QUIC client access control

File: docset/winserver2025-ps/smbshare/Grant-SmbClientAccessToServer.md [hunk 23]
```diff
@@ -0,0 +1,289 @@
...
+ms.date: 02/22/2024
+online version: https://learn.microsoft.com/powershell/module/smbshare/grant-smbclientaccesstoserver?view=windowsserver2025-ps&wt.mc_id=ps-gethelp
+schema: 2.0.0
+title: Grant-SmbClientAccessToServer
+---
+
+# Grant-SmbClientAccessToServer
+
+## SYNOPSIS
+Grants SMB client access to a specified server.
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/Grant-SmbClientAccessToServer.md:14]: for SMB over QUIC client access control

File: docset/winserver2025-ps/smbshare/Revoke-SmbClientAccessToServer.md [hunk 24]
```diff
@@ -0,0 +1,262 @@
...
+description: Use this topic to help manage Windows and Windows Server technologies with Windows PowerShell.
+external help file: SmbClientAccessToServer.cdxml-help.xml
+Module Name: SmbShare
+ms.date: 02/22/2024
+online version: https://learn.microsoft.com/powershell/module/smbshare/revoke-smbclientaccesstoserver?view=windowsserver2025-ps&wt.mc_id=ps-gethelp
+schema: 2.0.0
+title: Revoke-SmbClientAccessToServer
+---
+
+# Revoke-SmbClientAccessToServer
```
Comment (nedpyle) [inline docset/winserver2025-ps/smbshare/Revoke-SmbClientAccessToServer.md:11]: I want to add better explanation of revoke vs grant vs block vs unblock, will get back to you

**Review by nedpyle — CHANGES_REQUESTED** [review]
more to come

**Pull request merged**